pallets/messages/src/lib.rs

```diff
@@ -385,4 +397,25 @@ impl<T: Config> Pallet<T> {
 			.map(|msg| msg.map_to_response(block_number_value, schema_payload_location))
 			.collect()
 	}
+
+	/// Validates a CID to conform to IPFS CIDv1 (or higher) formatting (does not validate decoded CID fields)
+	///
+	/// # Errors
+	/// * [`Error::UnsupportedCidVersion`] - CID version is not supported (V0)
+	/// * [`Error::InvalidCid`] - Unable to parse provided CID
+	///
+	pub fn validate_cid(in_cid: Vec<u8>) -> Result<Vec<u8>, DispatchError> {
+		// Decode SCALE encoded CID into string slice
+		let ba: &[u8] = &in_cid;
+
+		let cstr: &str = sp_std::str::from_utf8(ba).map_err(|_| Error::<T>::InvalidCid)?;
+		ensure!(cstr.len() > 2, Error::<T>::InvalidCid);
+		ensure!(&cstr[0..2] != "Qm", Error::<T>::UnsupportedCidVersion);
+
+		// Assume it's a multibase-encoded string. Decode it to a byte array so we can parse the CID.
+		let cid_b = multibase::decode(cstr).map_err(|_| Error::<T>::InvalidCid)?.1;
+		ensure!(Cid::read_bytes(&cid_b as &[u8]).is_ok(), Error::<T>::InvalidCid);
```

Comment:
```suggestion
		ensure!(Cid::read_bytes(&cid_b[..]).is_ok(), Error::<T>::InvalidCid);
```

Reply:
We are more accustom to suggested syntax than directly using `as` for conversion

Reply:
Fixed.